wigwam/commands.py

```diff
@@ -252,6 +255,64 @@ def cmake_install(tag: str, base: str, no_cache: bool = False) -> Image:
     )
 
 
+def test(
+    tag: str,
+    logfile: os.PathLike[str] | str,
+    compress_output: bool,
+    quiet_fail: bool,
+) -> None:
+    """
+    Run all ctests from the Docker image work directory.
+
+    Parameters
+    ----------
+    tag : str
+        The tag of the image to test.
+    logfile : os.PathLike[str] | str
+        The name of the XML test output file.
+    compress_output : bool
+        If True, compress the output of the test.
+    quiet_fail : bool
+        If True, don't output on failure.
+    """
+
+    prefixed_tag: str = prefix_image_tag(tag)
+
+    host_volume_path = "./Testing/Temps"
+    image_volume_path = "/tmp/Testing"
+    image: Image = Image(prefixed_tag)
+
+    move_cmd = ["cd", str(build_prefix())]
+    test_cmd = ["(", "ctest"]
+
+    # Add arguments
+    if not compress_output:
+        test_cmd += ["--no-compress-output"]
+    if not quiet_fail:
+        test_cmd += ["--output-on-failure"]
+
+    test_cmd += ["-T", "Test", "||", "true", ")"]
+    file_cmd = [
+        "cp",
+        f"{build_prefix()}/Testing/*/Test.xml",
+        f"{image_volume_path}/{logfile}",
+    ]
+
+    cmd: list[str] = move_cmd + ["&&"] + test_cmd + ["&&"] + file_cmd
+
+    command = " ".join(cmd)
```

Comment:
Give this a try and see if it causes any issues.

```suggestion
    command = shlex.join(cmd)
```